tools/ci_build/build.py

```diff
@@ -1216,12 +1216,12 @@ def run_adb_shell(cmd):
 
 
 def run_ios_tests(args, source_dir, config, cwd):
-    cpr = run_subprocess(["xcodebuild", "test", "-project", "./onnxruntime.xcodeproj",
+    cpr = run_subprocess(["xcodebuild", "test-without-building", "-project", "./onnxruntime.xcodeproj",
```

Comment:
Change from `test` => `test-without-building` will speedup the CI run by 50%
- [example run](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=363886&view=logs&j=f7cc61a9-cc70-56e7-b06c-4668ca17e426&t=16d281b5-1bfd-5309-f274-36d0dffd9cb1) with this change, note it has both CPU and CoreML steps, the CPU step took about 42 mins,
- [example run](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=364115&view=logs&j=f7cc61a9-cc70-56e7-b06c-4668ca17e426) without this change, the CPU step alone took about 89mins.